### ISSUE TITLE
Verilog compilation issues

### ISSUE THREAD

**Issue by arch-linux** [maboytim/TRS-80_Display_Adapter#2]
Hello! @maboytim,

I had the boards made, and I am attempting to write the FPGA, I am using VSCODE to compile your FPGA code. 

Using this guide [here](https://learn.lushaylabs.com/getting-setup-with-the-tang-nano-9k/)

I am having issues compiling due to a few issues. I am getting the feeling maybe you used a different toolkit or different software entirely for this.

`Processing vis80.lushay.json
Starting FPGA Toolchain
Starting Yosys CST Checking
     /----------------------------------------------------------------------------\
     |  yosys -- Yosys Open SYnthesis Suite                                       |
     |  Copyright (C) 2012 - 2024  Claire Xenia Wolf <claire@yosyshq.com>         |
     |  Distributed under an ISC-like license, type "license" to see terms        |
     \----------------------------------------------------------------------------/
    Parsing gowin_clkdiv.v
        - Module Gowin_CLKDIV parsed
    Parsing gowin_dpb.v
        - Module Gowin_DPB parsed
    Parsing gowin_rpll.v
        - Module Gowin_rPLL parsed
    Parsing BUFG.v
        - Module BUFG parsed
    Error: /Users/archlinuxusa/Downloads/TRS-80_Display_Adapter-main/fpga/vid80-3/src/top.v:47: ERROR: syntax error, unexpected ';', expecting '(' or '['
    Check lines 46-47 of file top.v you may be missing a semicolon or left a block open
Finished CST Checking
Task finished with errors exiting
Toolchain Completed
`

Could you share what you used to program the TANG Nano 9k? 
Did you use the VSCODE Chain of tools or  GOWIN?

Thank you very much.

**Comment by maboytim** [comment]
Hi,
I use the Gowin IDE.  I've not tried the open source tools.
There are two versions of the Gowin IDE - a full version and a free educational version.  I use the free educational version.
From your errors it looks like the compiler is having trouble with the Gowin IP blocks.  Since I don't know the open source tools I can't help you with that but you might ask someone in that community for help.  These blocks are really hard IP's to access resources on the chip like PLL's and BRAMs.
You shouldn't have to compile though unless you want to make changes.  The .fs programming file is in the impl/pnr folder which is all you really need.  I have used the openFPGALoader which works.  There is also a standalone programmer available from Gowin.  You can program to SRAM but that is volatile and will be lost if you cycle power.  To be persistent you need to program the embedded flash.
If you have any problems feel free to email me directly at ***@***.***
Matt

    On Friday, November 8, 2024 at 06:06:06 PM EST, Christopher Allen ***@***.***> wrote:  
 
 


Hello! @maboytim,

I had the boards made, and I am attempting to write the FPGA, I am using VSCODE to compile your FPGA code.

Using this guide here

I am having issues compiling due to a few issues. I am getting the feeling maybe you used a different toolkit or different software entirely for this.

Processing vis80.lushay.json Starting FPGA Toolchain Starting Yosys CST Checking /----------------------------------------------------------------------------\ | yosys -- Yosys Open SYnthesis Suite | | Copyright (C) 2012 - 2024 Claire Xenia Wolf ***@***.***> | | Distributed under an ISC-like license, type "license" to see terms | \----------------------------------------------------------------------------/ Parsing gowin_clkdiv.v - Module Gowin_CLKDIV parsed Parsing gowin_dpb.v - Module Gowin_DPB parsed Parsing gowin_rpll.v - Module Gowin_rPLL parsed Parsing BUFG.v - Module BUFG parsed Error: /Users/archlinuxusa/Downloads/TRS-80_Display_Adapter-main/fpga/vid80-3/src/top.v:47: ERROR: syntax error, unexpected ';', expecting '(' or '[' Check lines 46-47 of file top.v you may be missing a semicolon or left a block open Finished CST Checking Task finished with errors exiting Toolchain Completed 

Could you share what you used to program the TANG Nano 9k?
Did you use the VSCODE Chain of tools or GOWIN?

Thank you very much.

**Comment by arch-linux** [comment]
@maboytim 
Myth Confirmed!

After following your info, I got this going and it works flawlessly.
Thank you very much.

I have 9 other boards that I could potentially populate and produce. Before making any decisions, I wanted to confirm with your unlicense in the github that there are no issues with me selling the remaining PCBs. I want to respect your hard work. 

Its also acceptable if you want to speak about this over email - let me know and ill provide mine.

Respectfully

**Comment by maboytim** [comment]
I think when you create a project github makes you pick a license.  I tried to pick the most permissive license.  I certainly appreciate your concern of the developer but all of my projects are completely free to do with whatever you want with no credit to me whatsoever.  I'm glad you got it working.  Try PM'ing me for my email.
Matt
    On Friday, November 8, 2024 at 08:54:35 PM EST, Christopher Allen ***@***.***> wrote:  
 
 


@maboytim
Myth Confirmed!

After following your info, I got this going and it works flawlessly.
Thank you very much.

I have 9 other boards that I could potentially populate and produce. Before making any decisions, I wanted to confirm with your unlicense in the github that there are no issues with me selling the remaining PCBs. I want to respect your hard work.

Its also acceptable if you want to speak about this over email - let me know and ill provide mine.

Respectfully
